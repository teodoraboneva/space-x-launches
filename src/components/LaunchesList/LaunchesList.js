import Grid from "@material-ui/core/Grid";
import MediaCard from "../MediaCard/MediaCard";

const LaunchesList = ({ launchesData }) => {
  return (
    <Grid container spacing={3}>
      {launchesData.length === 0
        ? "No results"
        : launchesData.map(
            ({
              flight_number,
              // launch_landing,
              launch_success,
              launch_year,
              links: { mission_patch },
              mission_id,
              mission_name,
              rocket,
            }) => {
              const land_success = rocket.first_stage.cores[0].land_success;
              return (
                <Grid key={flight_number} item xs={12} sm={6} md={4} lg={3}>
                  <MediaCard
                    image={mission_patch}
                    heading={`${mission_name} #${flight_number}`}
                  >
                    <p>Mission Ids:</p>
                    {mission_id.length > 0 && (
                      <ul>
                        {mission_id.map((id) => (
                          <li key={id}>{id}</li>
                        ))}
                      </ul>
                    )}
                    <p>Launch Year: {launch_year}</p>
                    <p>Successful Launch: {launch_success?.toString()}</p>
                    <p>Successful Landing: {land_success?.toString()}</p>
                  </MediaCard>
                </Grid>
              );
            }
          )}
    </Grid>
  );
};

export default LaunchesList;
