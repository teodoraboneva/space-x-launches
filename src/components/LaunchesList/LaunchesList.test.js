import renderer from "react-test-renderer";
import LaunchesList from "./LaunchesList";

const launchesData = [
    {
      "flight_number": 1,
      "mission_name": "FalconSat",
      "mission_id": [],
      "launch_year": "2006",
      "rocket": {
        "first_stage": {
          "cores": [
            {
              "land_success": null,
            }
          ]
        },
        
      },
      "launch_success": false,
      "links": {
        "mission_patch": "https://images2.imgbox.com/40/e3/GypSkayF_o.png",
      },
    },
    {
      "flight_number": 7,
      "mission_name": "COTS 1",
      "mission_id": [
        "EE86F74"
      ],
      "launch_year": "2010",
      "rocket": {
        "first_stage": {
          "cores": [
            {
              "land_success": null,
            }
          ]
        },
      },
      "launch_success": true,
      "links": {
        "mission_patch": "https://images2.imgbox.com/00/2f/FhtEd0nB_o.png",
      },
    },
  ];

it("renders correctly", () => {
  const tree = renderer.create(<LaunchesList launchesData={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly with data", () => {
  const tree = renderer.create(<LaunchesList launchesData={launchesData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
