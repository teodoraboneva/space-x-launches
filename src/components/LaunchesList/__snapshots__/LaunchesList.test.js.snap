// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
>
  No results
</div>
`;

exports[`renders correctly with data 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
  >
    <div
      className="MuiPaper-root MuiCard-root media-card MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiCardMedia-root"
          style={
            Object {
              "backgroundImage": "url(\\"https://images2.imgbox.com/40/e3/GypSkayF_o.png\\")",
            }
          }
          title=""
        />
        <div
          className="MuiCardContent-root"
        >
          <h2
            className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            FalconSat #1
          </h2>
          <div
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            <p>
              Mission Ids:
            </p>
            <p>
              Launch Year: 
              2006
            </p>
            <p>
              Successful Launch: 
              false
            </p>
            <p>
              Successful Landing: 
            </p>
          </div>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
  >
    <div
      className="MuiPaper-root MuiCard-root media-card MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        className="MuiButtonBase-root MuiCardActionArea-root"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <div
          className="MuiCardMedia-root"
          style={
            Object {
              "backgroundImage": "url(\\"https://images2.imgbox.com/00/2f/FhtEd0nB_o.png\\")",
            }
          }
          title=""
        />
        <div
          className="MuiCardContent-root"
        >
          <h2
            className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            COTS 1 #7
          </h2>
          <div
            className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            <p>
              Mission Ids:
            </p>
            <ul>
              <li>
                EE86F74
              </li>
            </ul>
            <p>
              Launch Year: 
              2010
            </p>
            <p>
              Successful Launch: 
              true
            </p>
            <p>
              Successful Landing: 
            </p>
          </div>
        </div>
        <span
          className="MuiCardActionArea-focusHighlight"
        />
      </button>
    </div>
  </div>
</div>
`;
